### PR TITLE
MaterialKey: expose constrainMaterial to Android

### PR DESCRIPTION
**Pull request by alexey-pelykh**
cc @romainguy

### CODE REVIEW

File: android/gltfio-android/src/main/cpp/MaterialKey.cpp [hunk 1]
```diff
@@ -72,6 +72,32 @@ Java_com_google_android_filament_gltfio_MaterialProvider_00024MaterialKey_nGloba
...
+    auto& helper = MaterialKeyHelper::get();
+    helper.copy(env, nativeMaterialKey, materialKey);
+
+    UvMap nativeUvMap = {};
+    constrainMaterial(&nativeMaterialKey, &nativeUvMap);
+
+    // Copy the UvMap results from the native array into the JVM array.
+    jint* elements = env->GetIntArrayElements(uvMap, nullptr);
+    if (elements) {
+        const size_t javaSize = env->GetArrayLength(uvMap);
```
Comment (romainguy) [inline android/gltfio-android/src/main/cpp/MaterialKey.cpp:90]: Use `std::copy_n` instead. For example:

```
std::copy_n(asset->getEntities(),
            std::min(available, (jsize) asset->getEntityCount()), entities);
```

File: android/gltfio-android/src/main/java/com/google/android/filament/gltfio/MaterialProvider.java [hunk 2]
```diff
@@ -70,6 +70,12 @@ public MaterialKey() {}
             nGlobalInit();
         }
         private static native void nGlobalInit();
+
+        public void constrainMaterial(@NonNull @Size(min = 8) int[] uvmap) {
```
Comment (romainguy) [inline android/gltfio-android/src/main/java/com/google/android/filament/gltfio/MaterialProvider.java:74]: Please keep the static native methods last like we do in the other JNI bindings. Also, could you add a documentation block to explain what this API is for?

**Review by romainguy — CHANGES_REQUESTED** [review]

File: android/gltfio-android/src/main/cpp/MaterialKey.cpp (hunk 1 above)
Comment (alexey-pelykh) [reply android/gltfio-android/src/main/cpp/MaterialKey.cpp:90]: Sure, but I've copied that snippet from the other part of Filament, literally neighboring file...

File: android/gltfio-android/src/main/java/com/google/android/filament/gltfio/MaterialProvider.java (hunk 2 above)
Comment (alexey-pelykh) [reply android/gltfio-android/src/main/java/com/google/android/filament/gltfio/MaterialProvider.java:74]: `constrainMaterial` has an explanation in native code, it's just a wrapper. Should I copy it over?

Comment (romainguy) [reply android/gltfio-android/src/main/java/com/google/android/filament/gltfio/MaterialProvider.java:74]: That's usually what we do yes. More importantly the doc should mention why a user of gltfio should care or use this API.

**Review by romainguy — APPROVED** [review]
Thank you!

File: android/gltfio-android/src/main/cpp/MaterialKey.cpp (hunk 1 above)
Comment (alexey-pelykh) [reply android/gltfio-android/src/main/cpp/MaterialKey.cpp]: Ugh, copy_n won't work - types are different :|

Comment (romainguy) [reply android/gltfio-android/src/main/cpp/MaterialKey.cpp]: Argh, sorry about that. That explains the original code :(

**Review by romainguy — APPROVED** [review]

**Pull request merged**